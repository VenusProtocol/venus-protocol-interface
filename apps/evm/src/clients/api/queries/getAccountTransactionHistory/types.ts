import type { ChainId, VToken } from '@venusprotocol/chains';
import type { Token, TxType } from 'types';
import type { Address } from 'viem';
import type { GetPoolsOutput } from '../useGetPools/types';

export interface ApiAccountHistoricalTransaction {
  id: string;
  txHash: string;
  txIndex: number;
  txTimestamp: Date;
  blockNumber: string;
  txType: TxType;
  accountAddress: Address;
  contractAddress: Address;
  amountVTokenMantissa: string | null;
  amountUnderlyingMantissa: string | null;
  underlyingAddress: string;
  underlyingTokenPriceMantissa: string | null;
  chainId: ChainId;
}

export interface AmountTransaction {
  txType: TxType;
  hash: string;
  blockTimestamp: Date;
  blockNumber: string;
  accountAddress: string;
  contractAddress: string;
  chainId: ChainId;
  poolName: string;
  vTokenSymbol: string;
  amountTokens: BigNumber | undefined;
  amountCents: BigNumber | undefined;
  token: Token | VToken;
}

export interface AccountTransactionHistoryApiResponse {
  count: number;
  results: ApiAccountHistoricalTransaction[];
}

export interface GetAccountTransactionHistoryInput {
  accountAddress: string;
  contractAddress: string;
  chainId: ChainId;
  getPoolsData: GetPoolsOutput | undefined;
  type?: number;
  page?: number;
}

export type GetAccountTransactionHistoryOutput = {
  count: number;
  transactions: AmountTransaction[];
};
