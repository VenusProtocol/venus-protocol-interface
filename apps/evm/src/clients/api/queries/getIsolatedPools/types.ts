import type { PoolLens, Prime, ResilientOracle } from 'libs/contracts';
import type { Provider } from 'libs/wallet';
import type { ChainId, Pool, Token } from 'types';

export interface GetIsolatedPoolsInput {
  chainId: ChainId;
  xvs: Token;
  tokens: Token[];
  provider: Provider;
  poolRegistryContractAddress: string;
  vTreasuryContractAddress: string;
  poolLensContract: PoolLens;
  resilientOracleContract: ResilientOracle;
  blocksPerDay?: number;
  primeContract?: Prime;
  accountAddress?: string;
}

export interface GetIsolatedPoolsOutput {
  pools: Pool[];
}
