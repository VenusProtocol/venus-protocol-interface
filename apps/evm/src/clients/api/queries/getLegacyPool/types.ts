import {
  LegacyPoolComptroller,
  Prime,
  ResilientOracle,
  VaiController,
  VenusLens,
} from 'packages/contracts';
import { Pool, Token } from 'types';

export interface GetLegacyPoolInput {
  blocksPerDay: number;
  name: string;
  description: string;
  xvs: Token;
  vai: Token;
  tokens: Token[];
  legacyPoolComptrollerContract: LegacyPoolComptroller;
  venusLensContract: VenusLens;
  resilientOracleContract: ResilientOracle;
  vaiControllerContract: VaiController;
  primeContract?: Prime;
  accountAddress?: string;
}

export interface GetLegacyPoolOutput {
  pool: Pool;
}
