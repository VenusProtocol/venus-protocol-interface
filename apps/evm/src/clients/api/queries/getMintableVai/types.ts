import BigNumber from 'bignumber.js';

import { Vai, VaiController } from 'libs/contracts';

export interface GetMintableVaiInput {
  accountAddress: string;
  vaiControllerContract: VaiController;
  vaiContract: Vai;
}

export interface GetMintableVaiOutput {
  vaiLiquidityMantissa: BigNumber;
  accountMintableVaiMantissa: BigNumber;
}
