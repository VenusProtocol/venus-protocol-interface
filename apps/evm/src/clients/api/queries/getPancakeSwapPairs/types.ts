import { Pair as PSPair, Token as PSToken } from '@pancakeswap/sdk/dist/index.js';
import BigNumber from 'bignumber.js';

import { type Provider } from 'libs/wallet';
import { PSTokenCombination } from 'types';

export interface GetPancakeSwapPairsInput {
  provider: Provider;
  tokenCombinations: PSTokenCombination[];
}

export interface PairAddress {
  tokenCombination: PSTokenCombination;
  address: string;
}

export interface TokenReserve {
  token: PSToken;
  reserveMantissa: BigNumber;
}

export type GetPancakeSwapPairsOutput = {
  pairs: PSPair[];
};
