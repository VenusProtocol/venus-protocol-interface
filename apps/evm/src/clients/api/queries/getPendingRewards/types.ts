import BigNumber from 'bignumber.js';

import { PoolLens, Prime, ResilientOracle, VaiVault, VenusLens, XvsVault } from 'libs/contracts';
import { Token } from 'types';

export interface GetPendingRewardsInput {
  tokens: Token[];
  isolatedPoolComptrollerAddresses: string[];
  xvsVestingVaultPoolCount: number;
  accountAddress: string;
  resilientOracleContract: ResilientOracle;
  poolLensContract: PoolLens;
  xvsVaultContract: XvsVault;
  vaiVaultContract?: VaiVault;
  venusLensContract?: VenusLens;
  primeContract?: Prime;
  legacyPoolComptrollerContractAddress?: string;
}

export interface GetPendingRewardsOutput {
  pendingRewardGroups: PendingRewardGroup[];
}

export interface IsolatedPoolPendingReward {
  rewardToken: Token;
  rewardAmountMantissa: BigNumber;
  rewardAmountCents: BigNumber | undefined;
  vTokenAddressesWithPendingReward: string[];
  rewardsDistributorAddress: string;
}

export interface IsolatedPoolPendingRewardGroup {
  type: 'isolatedPool';
  comptrollerAddress: string;
  pendingRewards: IsolatedPoolPendingReward[];
}

export interface LegacyPoolPendingRewardGroup {
  type: 'legacyPool';
  comptrollerAddress: string;
  rewardToken: Token;
  rewardAmountMantissa: BigNumber;
  rewardAmountCents: BigNumber | undefined;
  vTokenAddressesWithPendingReward: string[];
}

export interface VaultPendingRewardGroup {
  type: 'vault';
  isDisabled: boolean;
  stakedToken: Token;
  rewardToken: Token;
  rewardAmountMantissa: BigNumber;
  rewardAmountCents: BigNumber | undefined;
}

export interface XvsVestingVaultPendingRewardGroup {
  type: 'xvsVestingVault';
  isDisabled: boolean;
  poolIndex: number;
  stakedToken: Token;
  rewardToken: Token;
  rewardAmountMantissa: BigNumber;
  rewardAmountCents: BigNumber | undefined;
}

export interface PrimePendingReward {
  rewardToken: Token;
  rewardAmountMantissa: BigNumber;
  rewardAmountCents: BigNumber | undefined;
}

export interface PrimePendingRewardGroup {
  type: 'prime';
  isDisabled: boolean;
  vTokenAddressesWithPendingReward: string[];
  pendingRewards: PrimePendingReward[];
}

export type PendingRewardGroup =
  | LegacyPoolPendingRewardGroup
  | IsolatedPoolPendingRewardGroup
  | VaultPendingRewardGroup
  | XvsVestingVaultPendingRewardGroup
  | PrimePendingRewardGroup;
