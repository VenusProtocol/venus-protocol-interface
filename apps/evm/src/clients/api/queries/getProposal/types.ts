import type { Proposal, ProposalState, ProposalType, VoteSupport } from 'types';

export interface GetProposalInput {
  proposalId: number | string;
  accountAddress: string | undefined;
}

export interface ProposalActionApiResponse {
  actionIndex: number;
  calldata: string;
  signature: string;
  target: string;
  value: string | null;
}

type VoterApiResponse = {
  proposalId: number;
  address: string;
  blockNumber: number;
  blockTimestamp: number;
  reason: string | undefined;
  votesMantissa: string;
  support: VoteSupport;
};

export interface ProposalApiResponse {
  abstainedVotesMantissa: string;
  againstVotesMantissa: string;
  cancelBlock: number | null;
  cancelTimestamp: number | null;
  cancelTxHash: string | null;
  createdBlock: number | null;
  createdTimestamp: number | null;
  createdTxHash: string | null;
  description: string;
  endBlock: number;
  endTimestamp: number;
  eta: number | null;
  executedBlock: number | null;
  executedTimestamp: number | null;
  executedTxHash: string | null;
  forVotesMantissa: string;
  governorName: string;
  proposer: string;
  queuedBlock: number | null;
  queuedTimestamp: number | null;
  queuedTxHash: string | null;
  startBlock: number;
  startTimestamp: number;
  state: ProposalState;
  proposalType: ProposalType;
  proposalId: number;
  proposalActions: ProposalActionApiResponse[] | undefined;
  votes: VoterApiResponse[] | undefined;
}

export type GetProposalOutput = Proposal;
