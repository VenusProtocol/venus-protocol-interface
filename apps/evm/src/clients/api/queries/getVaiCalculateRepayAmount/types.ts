import BigNumber from 'bignumber.js';

import { VaiController } from 'packages/contracts';

export interface GetVaiCalculateRepayAmountInput {
  accountAddress: string;
  repayAmountMantissa: BigNumber;
  vaiControllerContract: VaiController;
}

export type GetVaiCalculateRepayAmountOutput = {
  vaiRepayAmountAfterFeeMantissa: BigNumber;
  vaiCurrentInterestMantissa: BigNumber;
  vaiPastInterestMantissa: BigNumber;
  vaiTotalInterestMantissa: BigNumber;
  feePercentage: number;
};
