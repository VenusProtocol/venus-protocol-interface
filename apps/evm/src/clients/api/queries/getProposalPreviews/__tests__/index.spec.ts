import fakeAccountAddress from '__mocks__/models/address';
import BigNumber from 'bignumber.js';
import { getProposalPreviews as getGqlProposalPreviews } from 'clients/subgraph';
import { ChainId, ProposalState } from 'types';
import type Vi from 'vitest';
import { type GetProposalPreviewsInput, getProposalPreviews } from '..';

const fakeParams: GetProposalPreviewsInput = {
  chainId: ChainId.BSC_TESTNET,
  currentBlockNumber: 1,
  proposalMinQuorumVotesMantissa: new BigNumber(10),
  proposalExecutionGracePeriodMs: 1209600000,
  blockTimeMs: 3000,
  accountAddress: fakeAccountAddress,
};

describe('getProposalPreviews', () => {
  beforeEach(() => {
    vi.useFakeTimers().setSystemTime(new Date(1710401645000));
  });

  it('returns proposal previews in the correct format', async () => {
    const res = await getProposalPreviews(fakeParams);

    expect(res).toMatchSnapshot();
  });

  it('sets "where" parameter correctly based on passed "proposalState" parameter', async () => {
    const proposalStates = Object.values(ProposalState).filter(
      (value): value is ProposalState => !Number.isNaN(+value),
    );

    for (let i = 0; i < proposalStates.length; i++) {
      const proposalState = proposalStates[i];

      await getProposalPreviews({
        ...fakeParams,
        proposalState,
      });

      expect((getGqlProposalPreviews as Vi.Mock).mock.calls[i][0]).toMatchSnapshot();
    }
  });

  it('sets "where" parameter correctly based on passed "search" parameter', async () => {
    await getProposalPreviews({
      ...fakeParams,
      search: 'fake search',
    });

    expect((getGqlProposalPreviews as Vi.Mock).mock.calls[0][0]).toMatchSnapshot();
  });
});
