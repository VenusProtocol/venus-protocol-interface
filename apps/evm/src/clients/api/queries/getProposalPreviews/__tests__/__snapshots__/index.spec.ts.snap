// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getProposalPreviews > returns proposal previews in the correct format 1`] = `
{
  "proposalPreviews": [
    {
      "abstainedVotesMantissa": "1.05462e+23",
      "againstVotesMantissa": "2.05462e+23",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
    This VIP does the following:
    1. Updates the implementation of all VTokens and Comptroller market facet in Core Pool
    2. Updates the implementation of VTokens and Comptrollers in IL
    3. Accepts the ownership of the NativeTokenGateway contract
    4. Gives call permissions to Timelocks for seizeVenus function
    5. Sets XVS and vXVS address in unitroller",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-Gateway Update VToken and comptroller implementation in IL and Core Pool and introduces the NativeTokenGateway contract",
        "version": "v2",
      },
      "endDate": 2027-11-13T07:27:47.000Z,
      "etaDate": 2024-03-14T07:34:08.000Z,
      "executedDate": 2024-03-14T07:38:32.000Z,
      "forVotesMantissa": "6.05462e+23",
      "proposalId": 387,
      "proposalType": 0,
      "queuedDate": 2024-03-14T07:24:08.000Z,
      "state": 0,
      "userVoteSupport": 1,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
    This VIP does the following:
    1. Updates the implementation of all VTokens and Comptroller market facet in Core Pool
    2. Sets the AccessControlManager, ProtocolShareReserve and ReduceReservesBlockDelta in vTRX, vUST and vLUNA
    3. Updates the implementation of VTokens and Comptrollers in IL
    4. Accepts the ownership of the NativeTokenGateway contract",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-Gateway Update VToken and comptroller implementation in IL and Core Pool and introduces the NativeTokenGateway contract",
        "version": "v2",
      },
      "endDate": 2027-11-05T13:32:44.000Z,
      "etaDate": 2024-03-06T13:33:52.000Z,
      "executedDate": 2024-03-06T13:38:28.000Z,
      "forVotesMantissa": "6.05462e+23",
      "proposalId": 386,
      "proposalType": 0,
      "queuedDate": 2024-03-06T13:23:52.000Z,
      "state": 0,
      "userVoteSupport": 1,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
        This VIP upgrades the implementation of RiskFundConverter and SingleTokenConverter.",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-Converter2 upgrades the implementation of token converters",
        "version": "v2",
      },
      "endDate": 2027-10-29T09:19:29.000Z,
      "etaDate": 2024-02-28T09:06:22.000Z,
      "executedDate": 2024-02-28T09:15:01.000Z,
      "forVotesMantissa": "6.05462e+23",
      "proposalId": 385,
      "proposalType": 0,
      "queuedDate": 2024-02-28T08:56:22.000Z,
      "state": 0,
      "userVoteSupport": 1,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
        This VIP upgrades the implementation of RiskFundConverter and SingleTokenConverter.",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-Converter2 upgrades the implementation of token converters",
        "version": "v2",
      },
      "endDate": 2027-10-29T09:03:41.000Z,
      "etaDate": undefined,
      "executedDate": undefined,
      "forVotesMantissa": "0",
      "proposalId": 384,
      "proposalType": 0,
      "queuedDate": undefined,
      "state": 0,
      "userVoteSupport": undefined,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "#### Description
The borrow cap for WBETH on Venus Core Pool is currently at 100% utilization.

There is currently a single user borrowing 1.06k WBETH, amounting to 53% of the total borrows of the asset. The user is currently borrowing $8.19M (in WBETH and USDT) against his $13.2M BTCB collateral, bringing his health score to 1.29.

Utilizing our supply and borrow cap methodology, Chaos Labs recommends doubling the borrow cap, setting it at 4K WBETH.

#### References
* [VIP Simulation](https://github.com/VenusProtocol/vips/pull/214)
  ",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-260 WBETH (Core Pool) - increase borrow cap to 4,000",
        "version": "v2",
      },
      "endDate": 2027-10-23T20:39:26.000Z,
      "etaDate": 2024-02-22T20:25:43.000Z,
      "executedDate": 2024-02-22T20:28:43.000Z,
      "forVotesMantissa": "1.2054657949832e+24",
      "proposalId": 383,
      "proposalType": 0,
      "queuedDate": 2024-02-22T20:15:43.000Z,
      "state": 0,
      "userVoteSupport": 1,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds with this proposal",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "Configure VAI Price",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-260 Configure VAI Price",
        "version": "v2",
      },
      "endDate": 2027-10-23T15:34:05.000Z,
      "etaDate": 2024-02-22T15:20:24.000Z,
      "executedDate": 2024-02-22T15:22:42.000Z,
      "forVotesMantissa": "6.05462e+23",
      "proposalId": 382,
      "proposalType": 0,
      "queuedDate": 2024-02-22T15:10:24.000Z,
      "state": 0,
      "userVoteSupport": 1,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
The borrow cap for WBETH on Venus Core Pool is currently at 100% utilization.

There is currently a single user borrowing 1.06k WBETH, amounting to 53% of the total borrows of the asset. The user is currently borrowing $8.19M (in WBETH and USDT) against his $13.2M BTCB collateral, bringing his health score to 1.29.

Utilizing our supply and borrow cap methodology, Chaos Labs recommends doubling the borrow cap, setting it at 4K WBETH.

#### References
* [VIP Simulation](https://github.com/VenusProtocol/vips/pull/180)
  ",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-260 WBETH (Core Pool) - increase borrow cap to 4,000",
        "version": "v2",
      },
      "endDate": 2027-10-22T17:07:08.000Z,
      "etaDate": 2024-02-21T16:48:09.000Z,
      "executedDate": undefined,
      "forVotesMantissa": "7.00103849974598612445922e+23",
      "proposalId": 381,
      "proposalType": 1,
      "queuedDate": 2024-02-21T16:43:09.000Z,
      "state": 0,
      "userVoteSupport": undefined,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
The borrow cap for WBETH on Venus Core Pool is currently at 100% utilization.

There is currently a single user borrowing 1.06k WBETH, amounting to 53% of the total borrows of the asset. The user is currently borrowing $8.19M (in WBETH and USDT) against his $13.2M BTCB collateral, bringing his health score to 1.29.

Utilizing our supply and borrow cap methodology, Chaos Labs recommends doubling the borrow cap, setting it at 4K WBETH.

#### References
* [VIP Simulation](https://github.com/VenusProtocol/vips/pull/180)
  ",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-260 WBETH (Core Pool) - increase borrow cap to 4,000",
        "version": "v2",
      },
      "endDate": 2027-10-22T16:46:08.000Z,
      "etaDate": undefined,
      "executedDate": undefined,
      "forVotesMantissa": "3e+23",
      "proposalId": 380,
      "proposalType": 1,
      "queuedDate": undefined,
      "state": 0,
      "userVoteSupport": undefined,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
            1. This VIP upgrades the implementation of RiskFundConverter, SingleTokenConverter.
            2. This VIP sets the address of the PSR in the Liquidator contract.",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-Converter upgrades the implementation of token converter and sets the PSR address in liquidator contract.",
        "version": "v2",
      },
      "endDate": 2027-10-21T15:08:29.000Z,
      "etaDate": 2024-02-20T14:58:20.000Z,
      "executedDate": 2024-02-20T19:00:18.000Z,
      "forVotesMantissa": "6.05462e+23",
      "proposalId": 379,
      "proposalType": 0,
      "queuedDate": 2024-02-20T14:48:20.000Z,
      "state": 0,
      "userVoteSupport": 1,
    },
    {
      "abstainedVotesMantissa": "0",
      "againstVotesMantissa": "0",
      "cancelDate": undefined,
      "description": {
        "abstainDescription": "I am indifferent to whether Venus Protocol proceeds or not",
        "againstDescription": "I do not think that Venus Protocol should proceed with this proposal",
        "description": "
            This VIP upgrades the implementation of RiskFundConverter and SingleTokenConverter",
        "forDescription": "I agree that Venus Protocol should proceed with this proposal",
        "title": "VIP-Converter upgrades the implementation of token converter",
        "version": "v2",
      },
      "endDate": 2027-10-21T14:46:20.000Z,
      "etaDate": undefined,
      "executedDate": undefined,
      "forVotesMantissa": "0",
      "proposalId": 378,
      "proposalType": 0,
      "queuedDate": undefined,
      "state": 0,
      "userVoteSupport": undefined,
    },
  ],
  "total": 1,
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 1`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "startBlock_gt": 1,
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 2`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "endBlock_gt": 1,
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 3`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "canceled_not": null,
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 4`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "or": [
        {
          "canceled": null,
          "endBlock_lt": 1,
          "passing": false,
        },
        {
          "or": [
            {
              "canceled": null,
              "endBlock_lt": 1,
              "forVotes_lt": "10",
            },
          ],
        },
      ],
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 5`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "canceled": null,
      "endBlock_lt": 1,
      "executed": null,
      "forVotes_gte": "10",
      "passing": true,
      "queued": null,
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 6`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "canceled": null,
      "executed": null,
      "executionEta_gt": "1710401645000000",
      "queued_not": null,
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 7`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "canceled": null,
      "executed": null,
      "executionEta_lt": "1710401645000000",
      "queued_not": null,
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "proposalState" parameter 8`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "executed_not": null,
    },
  },
}
`;

exports[`getProposalPreviews > sets "where" parameter correctly based on passed "search" parameter 1`] = `
{
  "chainId": 97,
  "variables": {
    "accountAddress": "0x3d759121234cd36f8124c21afe1c6852d2bed848",
    "limit": 10,
    "skip": 0,
    "where": {
      "description_contains_nocase": "fake search",
    },
  },
}
`;
